### PR TITLE
Provide guidance on when maintainers will (not) approve a PR

### PR DESCRIPTION
**Pull request by tomkerkhove**
Provide guidance on when maintainers will (not) approve a PR

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)

Closes https://github.com/kedacore/governance/issues/94

### CODE REVIEW

**Review by JorTurFer — APPROVED** [review]
It's nice!
Maybe I'd also add something to clarify that all the checks should pass, but maintainers can merge the PRs without them if they think that it's okey.

As this governance applies to all the repos, I'd also add something clarifying that maintainers can merge PRs without approvals under their own criteria in case of necessity. Why should we add this? During the release process, sometimes it is  difficult to get ACKs. Other examples are tooling repos (test-tools or testing-infrastructure), in those repos doesn't make sense to wait for approvals (IMO)

**Comment by tomkerkhove** [comment]
> Maybe I'd also add something to clarify that all the checks should pass, but maintainers can merge the PRs without them if they think that it's okey.

Would you mind opening a proposal in the review please?

> As this governance applies to all the repos, I'd also add something clarifying that maintainers can merge PRs without approvals under their own criteria in case of necessity. Why should we add this? During the release process, sometimes it is difficult to get ACKs.

I'd argue if nobody is able to ACK, then we should hold off to continue though. Second pair of eyes is crucial to me.

> Other examples are tooling repos (test-tools or testing-infrastructure), in those repos doesn't make sense to wait for approvals (IMO)

Good point, care to open a proposal?

File: GOVERNANCE.md
Comment (JorTurFer) [inline GOVERNANCE.md:35]:
```suggestion
- It does not violate the [code of conduct](https://github.com/kedacore/governance/blob/main/CODE_OF_CONDUCT.md).

Apart from the conditions above, all the required checks in the PR have to pass. Maintainers still have the right to merge a PR with one or more failing checks if they, after reviewing the check, consider that it's an unrelated failure (like a transient failure on an e2e test).

In case of dispute, other maintainers are requested to review and agree what the best outcome is of a PR.
```

File: GOVERNANCE.md
Comment (JorTurFer) [inline GOVERNANCE.md:39]:
```suggestion
Once a single maintainer approves a PR, they can merge it unless they want to have a second pair of eyes. (optional)

This rules apply to all the repos whose code will impact directly in the end-users, like [`keda`](https://github.com/kedacore/keda), [`charts`](https://github.com/kedacore/charts), [`keda-docs`](https://github.com/kedacore/keda-docs), etc.
Other repositories used for internal tooling, like [`testing-infrastucture`](https://github.com/kedacore/testing-infrastructure), [`test-tools`](https://github.com/kedacore/test-tools), etc, are excluded from these rules and maintainers can merge any commit without requiring any other approval than themself (unless they want to have a second pair of eyes)

Exceptionally and always due to good and important reasons, maintainers can merge changes without requiring additional approval, but they must notify the other maintainers about the action.

### Becoming A Maintainer
```

**Comment by JorTurFer** [comment]
> I'd argue if nobody is able to ACK, then we should hold off to continue though. Second pair of eyes is crucial to me.
> 
> >

I have added this part too:
> Exceptionally and always due to good and important reasons, maintainers can merge changes without requiring additional approval, but they must notify the other maintainers about the action.

As we are "signing a contract" with the governance documentation, I prefer to keep the door open to edge cases that could happen in the future. 
Merging without any ACK is something exceptional, but I believe that not blocking the option for urgent or risky cases is worth.

**Comment by JorTurFer** [comment]
Kindly reminder: We still have this pending xD

File: GOVERNANCE.md
Comment (tomkerkhove) [reply GOVERNANCE.md:41]: I'm not sure if we need special rules, care to elaborate on this please?

**Comment by tomkerkhove** [comment]
@kedacore/keda-maintainers Any other reviewers?

**Review by JorTurFer — APPROVED** [review]

File: GOVERNANCE.md
Comment (JorTurFer) [reply GOVERNANCE.md:41]: Sure, xD
some repos, such as test-tool, testing-infrastructure  are repos where changes don't affect to public artifacts but where there can be multiple changes at once if we are spinning up something (for example, I needed to merge some commits generating the K6 runner image, or when there were terraform issues with managed azure prometheus). If I had to ask for a review on any change, it'd have been terrible.

I totally agree with having a review on repos with public impact, but for internal stuff I think that requiring an approval is something that slows the process unnecessarily.

About the last paragraph, I see this governance as "a contract that we are signing", I prefer to keep the door open for exceptional cases. we can rephrase it as: "Exceptional situations could have exceptional responses, notifying always to other maintainers". But I won't close all the doors for rare situations.

Maybe the problem is that I understand this as a working way that we must follow at 100% of the cases and it's more a genera guidelines about how we work

**Comment by tomkerkhove** [comment]
@zroubalik @jeffhollan @ahmelsayed PTAL

**Comment by JorTurFer** [comment]
We should merge this @tomkerkhove @zroubalik

**Review by zroubalik — APPROVED** [review]

**Comment by tomkerkhove** [comment]
@JorTurFer @zroubalik We have two approvals, OK to merge or were there changes I had to make? (I think we are OK but want to make sure)

**Comment by JorTurFer** [comment]
There is a super majority, 3 of 4 companies xD
Let's merge it

**Pull request merged**
